File: python/purehtml/transformers/types/html_transformer.py
from abc import ABC
from typing import List

from purehtml.transformers.transform_params import TransformParams
from purehtml.transformers.transformer import Transformer


class HTMLTransformer(Transformer, ABC):
    def __init__(self, args: List[str] = None):
        """
        Initialize the transformer with optional arguments.
        :param args: Arguments to configure the transformer (unused in this case).
        """
        super().__init__()
        self.args = args if args is not None else []

    def set_args(self, args: List[str]):
        """
        Set the list of attributes to retrieve.
        :param args: A list of attribute names.
        """
        self.args = args

    def get_name(self) -> str:
        """
        Return the name of the transformer.
        :return: The name of the transformer.
        """
        return "html"

    def transform(self, params: TransformParams) -> str:
        """
        Retrieve the HTML content of the given element.
        :param params: The parameters that contain the element to extract HTML content from.
        :return: The HTML content of the element.
        """
        element = params.get_element()

        # If element is found, return its HTML content; otherwise, return an empty string
        if element:
            return element.html()
        return ""

File: java/src/main/java/org/purescraps/purehtml/transformers/types/HTMLTransformer.java
package org.purescraps.purehtml.transformers.types;

import org.purescraps.purehtml.backend.PureHTMLNode;
import org.purescraps.purehtml.transformers.TransformParams;
import org.purescraps.purehtml.transformers.Transformer;

import java.util.List;

public class HTMLTransformer extends Transformer {
    @SuppressWarnings("unused")
    private List<String> args;

    @SuppressWarnings("unused")
    public HTMLTransformer(List<String> args) {
        this.args = args;
    }

    @SuppressWarnings("unused")
    public HTMLTransformer() {
        this.args = null;
    }

    public void setArgs(List<String> args) {
        this.args = args;
    }

    // Static method to return the name of the transformer
    public String getName() {
        return "html";
    }

    // The transform method that returns the HTML content of the element
    @Override
    public Object transform(TransformParams params) {
        // Retrieve elements from the TransformParams object
        PureHTMLNode elements = params.getElement();
        // If elements is not null, call html() method; otherwise, return null
        if (elements != null) {
            return elements.html(); // Assuming html() is a method in HTMLNode
        }
        return "";
    }
}
